visible_to/BUILD

```diff
@@ -0,0 +1,344 @@
+# Package groups defined for use in kubernetes visibility rules.
```

Comment:
Absolutely not the expectation.  I don't want to have a fine grained set of rules.

The goal of this step is to _identify_ the bad deps and stop new bad deps.  FYI, two new ones were introduced while I was putting this together.

Subsequent PR's will remove the bad deps.  This will remove the package groups labelled BAD.  Once that happens, the remaining groups conflate to larger, coarser package groups (as you say, "high level") that make sense and need little maintenance.

What I've done here is bought a giant mansion full of _junk_ and good stuff.  I've sorted the _junk_ into careful piles, and had to put the good stuff into somewhat arbitrary piles only to separate them from the junk.

Had you or anyone else done the work, you'd likely have arranged the good stuff differently.  It doesn't matter - we just want to get at the junk.  Once we get rid of the junk, we put the good stuff "back", and instead of listing all package dirs exhaustively, we can use ... to indicate everything below a tree.  For this intermediate step, i want explicit content so i can see what's what.

I could spend time and energy trying to collapse the good stuff now, but i don't care.  Once the bad package groups are removed, the good ones combine relatively easily.

As far as on-going maintenance goes - the BUILD file should only need editting when one is either creating, deleting or moving whole packages.  It's a coarse level thing.